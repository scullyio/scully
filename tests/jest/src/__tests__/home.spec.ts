import { readPage, replaceIndexNG } from '../test-config.helper';
import { expect } from '@jest/globals';

describe('Check list of all', () => {
  it('Check clean all list from scully', () => {
    const index: string = readPage('/home/all');
    const cleanIndex = replaceIndexNG(index);
    expect(cleanIndex).toMatchSnapshot();
  });
});
