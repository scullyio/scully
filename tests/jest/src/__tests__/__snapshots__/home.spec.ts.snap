// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check list of all Check clean all list from scully 1`] = `
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta name="generator">
    <meta charset="utf-8">
    <title>
      All routes in application
    </title>
    <base href="/">
    <meta name="viewport"
          content="width=device-width, initial-scale=1"
    >
    <link rel="icon"
          type="image/x-icon"
          href="favicon.ico"
    >
    <link rel="stylesheet"
          href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.18.1/styles/default.min.css"
    >
    <link rel="stylesheet"
          href="styles.css"
    >
    <style>
      []{display:grid;height:100vh;grid-template-rows:60px 1fr 60px}[] > header[]{display:grid;background-color:#4169e1;color:#f5f5f5;margin:0;padding:0 10px;grid-template-columns:1fr 40px;place-items:center center}footer[]   h3[], header[]   h1[]{margin:0}[] > footer[]{display:grid;background-color:#4169e1;color:#f5f5f5;margin:0;padding-right:10px;justify-content:right;align-content:center}main[]{background-color:#f5f5f5;color:#040c24;padding:10px}
    </style>
    <style>
      li[]{display:grid;grid-template-columns:200px 200px}a.btn[]{display:inline-block;padding:10px;margin:10px 3px;width:200px;height:40px;border-radius:4px;background-color:#4169e1;color:#fff;text-align:center;text-decoration:none;box-shadow:0 1px 3px rgba(0,0,0,.12),0 1px 2px rgba(0,0,0,.24);transition:all .3s cubic-bezier(.25,.8,.25,1)}a.btn[]:hover{box-shadow:0 14px 28px rgba(0,0,0,.25),0 6px 6px rgba(0,0,0,.22)}a.btn[]:active{box-shadow:0 1px 3px rgba(0,0,0,.12),0 1px 2px rgba(0,0,0,.24)}
    </style>
  </head>
  <body>
    <app-root class="foo"
              =""
    >
      Scully demo blog app!
      <small ="">
        rendering inside scully
      </small>
      <a =""
         href="/home/"
      >
        üè†
      </a>
      <main ="">
        <router-outlet ="">
        </router-outlet>
        <app-static ="">
          <h1 ="">
            All routes in application
          </h1>
          <a =""
             class="btn"
             href="/home/"
          >
            Top level routes only
          </a>
          <a =""
             class="btn"
             href="/home/unpublished/"
          >
            Unpublished routes
          </a>
          <ul ="">
            <li ="">
              <strong ="">
                routelink
              </strong>
              <strong ="">
                full page reload
              </strong>
            </li>
            <li ="">
              <a =""
                 href="/about/"
              >
                /about
              </a>
              <a =""
                 href="/about/"
              >
                /about
              </a>
            </li>
            <li ="">
              <a =""
                 href="/home/all/"
              >
                All routes in application
              </a>
              <a =""
                 href="/home/all/"
              >
                All routes in application
              </a>
            </li>
            <li ="">
              <a =""
                 href="/home/unpublished/"
              >
                Unpublished routes in application
              </a>
              <a =""
                 href="/home/unpublished/"
              >
                Unpublished routes in application
              </a>
            </li>
            <li ="">
              <a =""
                 href="/home/"
              >
                Toplevel routes in application
              </a>
              <a =""
                 href="/home/"
              >
                Toplevel routes in application
              </a>
            </li>
            <li ="">
              <a =""
                 href="/home/"
              >
                /home
              </a>
              <a =""
                 href="/home/"
              >
                /home
              </a>
            </li>
            <li ="">
              <a =""
                 href="/blog/"
              >
                /blog
              </a>
              <a =""
                 href="/blog/"
              >
                /blog
              </a>
            </li>
            <li ="">
              <a =""
                 href="/blog/2014/2/12/asdf/"
              >
                Deeply embedded
              </a>
              <a =""
                 href="/blog/2014/2/12/asdf/"
              >
                Deeply embedded
              </a>
            </li>
            <li ="">
              <a =""
                 href="/blog/%D0%A0%D1%83%D1%81%D1%81%D0%BA%D0%B8%D0%B9-%D1%81%D0%BB%D0%B0%D0%B3"
              >
                This is my post in Russian
              </a>
              <a =""
                 href="/blog/–†—É—Å—Å–∫–∏–π-—Å–ª–∞–≥/"
              >
                This is my post in Russian
              </a>
            </li>
            <li ="">
              <a =""
                 href="/blog/page-1/"
              >
                My first page
              </a>
              <a =""
                 href="/blog/page-1/"
              >
                My first page
              </a>
            </li>
            <li ="">
              <a =""
                 href="/blog/page-3/"
              >
                My third page
              </a>
              <a =""
                 href="/blog/page-3/"
              >
                My third page
              </a>
            </li>
            <li ="">
              <a =""
                 href="/blog/page-4/"
              >
                The Rainbow Cat, A free book out of the Guttenberg library
              </a>
              <a =""
                 href="/blog/page-4/"
              >
                The Rainbow Cat, A free book out of the Guttenberg library
              </a>
            </li>
            <li ="">
              <a =""
                 href="/blog/page-5/"
              >
                /blog/page-5
              </a>
              <a =""
                 href="/blog/page-5/"
              >
                /blog/page-5
              </a>
            </li>
            <li ="">
              <a =""
                 href="/blog/page-6/"
              >
                My sixth page
              </a>
              <a =""
                 href="/blog/page-6/"
              >
                My sixth page
              </a>
            </li>
            <li ="">
              <a =""
                 href="/blog/test/testing/"
              >
                My test sub page
              </a>
              <a =""
                 href="/blog/test/testing/"
              >
                My test sub page
              </a>
            </li>
            <li ="">
              <a =""
                 href="/content/one/"
              >
                /content/one
              </a>
              <a =""
                 href="/content/one/"
              >
                /content/one
              </a>
            </li>
            <li ="">
              <a =""
                 href="/content/two/"
              >
                /content/two
              </a>
              <a =""
                 href="/content/two/"
              >
                /content/two
              </a>
            </li>
            <li ="">
              <a =""
                 href="/content/tree/"
              >
                /content/tree
              </a>
              <a =""
                 href="/content/tree/"
              >
                /content/tree
              </a>
            </li>
            <li ="">
              <a =""
                 href="/content/four/"
              >
                /content/four
              </a>
              <a =""
                 href="/content/four/"
              >
                /content/four
              </a>
            </li>
            <li ="">
              <a =""
                 href="/content/five/"
              >
                /content/five
              </a>
              <a =""
                 href="/content/five/"
              >
                /content/five
              </a>
            </li>
            <li ="">
              <a =""
                 href="/user/"
              >
                /user
              </a>
              <a =""
                 href="/user/"
              >
                /user
              </a>
            </li>
            <li ="">
              <a =""
                 href="/user/1/"
              >
                /user/1
              </a>
              <a =""
                 href="/user/1/"
              >
                /user/1
              </a>
            </li>
            <li ="">
              <a =""
                 href="/user/2/"
              >
                /user/2
              </a>
              <a =""
                 href="/user/2/"
              >
                /user/2
              </a>
            </li>
            <li ="">
              <a =""
                 href="/user/1/post/1/"
              >
                /user/1/post/1
              </a>
              <a =""
                 href="/user/1/post/1/"
              >
                /user/1/post/1
              </a>
            </li>
            <li ="">
              <a =""
                 href="/user/1/post/2/"
              >
                /user/1/post/2
              </a>
              <a =""
                 href="/user/1/post/2/"
              >
                /user/1/post/2
              </a>
            </li>
            <li ="">
              <a =""
                 href="/user/1/post/3/"
              >
                /user/1/post/3
              </a>
              <a =""
                 href="/user/1/post/3/"
              >
                /user/1/post/3
              </a>
            </li>
            <li ="">
              <a =""
                 href="/user/1/post/4/"
              >
                /user/1/post/4
              </a>
              <a =""
                 href="/user/1/post/4/"
              >
                /user/1/post/4
              </a>
            </li>
            <li ="">
              <a =""
                 href="/user/1/post/5/"
              >
                /user/1/post/5
              </a>
              <a =""
                 href="/user/1/post/5/"
              >
                /user/1/post/5
              </a>
            </li>
            <li ="">
              <a =""
                 href="/user/1/post/6/"
              >
                /user/1/post/6
              </a>
              <a =""
                 href="/user/1/post/6/"
              >
                /user/1/post/6
              </a>
            </li>
            <li ="">
              <a =""
                 href="/user/1/post/7/"
              >
                /user/1/post/7
              </a>
              <a =""
                 href="/user/1/post/7/"
              >
                /user/1/post/7
              </a>
            </li>
            <li ="">
              <a =""
                 href="/user/1/post/8/"
              >
                /user/1/post/8
              </a>
              <a =""
                 href="/user/1/post/8/"
              >
                /user/1/post/8
              </a>
            </li>
            <li ="">
              <a =""
                 href="/user/1/post/9/"
              >
                /user/1/post/9
              </a>
              <a =""
                 href="/user/1/post/9/"
              >
                /user/1/post/9
              </a>
            </li>
            <li ="">
              <a =""
                 href="/user/1/post/10/"
              >
                /user/1/post/10
              </a>
              <a =""
                 href="/user/1/post/10/"
              >
                /user/1/post/10
              </a>
            </li>
            <li ="">
              <a =""
                 href="/user/2/post/11/"
              >
                /user/2/post/11
              </a>
              <a =""
                 href="/user/2/post/11/"
              >
                /user/2/post/11
              </a>
            </li>
            <li ="">
              <a =""
                 href="/user/2/post/12/"
              >
                /user/2/post/12
              </a>
              <a =""
                 href="/user/2/post/12/"
              >
                /user/2/post/12
              </a>
            </li>
            <li ="">
              <a =""
                 href="/user/2/post/13/"
              >
                /user/2/post/13
              </a>
              <a =""
                 href="/user/2/post/13/"
              >
                /user/2/post/13
              </a>
            </li>
            <li ="">
              <a =""
                 href="/user/2/post/14/"
              >
                /user/2/post/14
              </a>
              <a =""
                 href="/user/2/post/14/"
              >
                /user/2/post/14
              </a>
            </li>
            <li ="">
              <a =""
                 href="/user/2/post/15/"
              >
                /user/2/post/15
              </a>
              <a =""
                 href="/user/2/post/15/"
              >
                /user/2/post/15
              </a>
            </li>
            <li ="">
              <a =""
                 href="/user/2/post/16/"
              >
                /user/2/post/16
              </a>
              <a =""
                 href="/user/2/post/16/"
              >
                /user/2/post/16
              </a>
            </li>
            <li ="">
              <a =""
                 href="/user/2/post/17/"
              >
                /user/2/post/17
              </a>
              <a =""
                 href="/user/2/post/17/"
              >
                /user/2/post/17
              </a>
            </li>
            <li ="">
              <a =""
                 href="/user/2/post/18/"
              >
                /user/2/post/18
              </a>
              <a =""
                 href="/user/2/post/18/"
              >
                /user/2/post/18
              </a>
            </li>
            <li ="">
              <a =""
                 href="/user/2/post/19/"
              >
                /user/2/post/19
              </a>
              <a =""
                 href="/user/2/post/19/"
              >
                /user/2/post/19
              </a>
            </li>
            <li ="">
              <a =""
                 href="/user/2/post/20/"
              >
                /user/2/post/20
              </a>
              <a =""
                 href="/user/2/post/20/"
              >
                /user/2/post/20
              </a>
            </li>
            <li ="">
              <a =""
                 href="/demo/0/"
              >
                page number 0
              </a>
              <a =""
                 href="/demo/0/"
              >
                page number 0
              </a>
            </li>
            <li ="">
              <a =""
                 href="/demo/1/"
              >
                page number 1
              </a>
              <a =""
                 href="/demo/1/"
              >
                page number 1
              </a>
            </li>
            <li ="">
              <a =""
                 href="/demo/2/"
              >
                page number 2
              </a>
              <a =""
                 href="/demo/2/"
              >
                page number 2
              </a>
            </li>
            <li ="">
              <a =""
                 href="/demo/3/"
              >
                page number 3
              </a>
              <a =""
                 href="/demo/3/"
              >
                page number 3
              </a>
            </li>
            <li ="">
              <a =""
                 href="/demo/4/"
              >
                page number 4
              </a>
              <a =""
                 href="/demo/4/"
              >
                page number 4
              </a>
            </li>
            <li ="">
              <a =""
                 href="/slow/"
              >
                /slow
              </a>
              <a =""
                 href="/slow/"
              >
                /slow
              </a>
            </li>
            <li ="">
              <a =""
                 href="/basehref/"
              >
                /basehref
              </a>
              <a =""
                 href="/basehref/"
              >
                /basehref
              </a>
            </li>
            <li ="">
              <a =""
                 href="/basehref/rewritten/"
              >
                /basehref/rewritten
              </a>
              <a =""
                 href="/basehref/rewritten/"
              >
                /basehref/rewritten
              </a>
            </li>
            <li ="">
              <a =""
                 href="/basehref/removed/"
              >
                /basehref/removed
              </a>
              <a =""
                 href="/basehref/removed/"
              >
                /basehref/removed
              </a>
            </li>
            <li ="">
              <a =""
                 href="/manualIdle/"
              >
                /manualIdle
              </a>
              <a =""
                 href="/manualIdle/"
              >
                /manualIdle
              </a>
            </li>
            <li ="">
              <a =""
                 href="/noScript/"
              >
                /noScript
              </a>
              <a =""
                 href="/noScript/"
              >
                /noScript
              </a>
            </li>
            <li ="">
              <a =""
                 href="/"
              >
                /
              </a>
              <a =""
                 href="/"
              >
                /
              </a>
            </li>
            <li ="">
              <a =""
                 href="/exclude/present/"
              >
                /exclude/present
              </a>
              <a =""
                 href="/exclude/present/"
              >
                /exclude/present
              </a>
            </li>
            <li ="">
              <a =""
                 href="/test/fake1/"
              >
                /test/fake1
              </a>
              <a =""
                 href="/test/fake1/"
              >
                /test/fake1
              </a>
            </li>
            <li ="">
              <a =""
                 href="/test/fake2/"
              >
                /test/fake2
              </a>
              <a =""
                 href="/test/fake2/"
              >
                /test/fake2
              </a>
            </li>
            <li ="">
              <a =""
                 href="/content/hello/"
              >
                /content/hello
              </a>
              <a =""
                 href="/content/hello/"
              >
                /content/hello
              </a>
            </li>
            <li ="">
              <a =""
                 href="/content/there/"
              >
                /content/there
              </a>
              <a =""
                 href="/content/there/"
              >
                /content/there
              </a>
            </li>
            <li ="">
              <a =""
                 href="/rawRoute/"
              >
                /rawRoute
              </a>
              <a =""
                 href="/rawRoute/"
              >
                /rawRoute
              </a>
            </li>
          </ul>
        </app-static>
      </main>
      <footer ="">
        <h3 ="">
          Made with ‚ù§Ô∏è
          <strong ="">
            @HeroDevs
          </strong>
        </h3>
      </footer>
    </app-root>
  </body>
</html>
`;
